pkgs/development/python-modules/martian/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib, buildPythonPackage, fetchPypi, callPackage
+, setuptools, zc_buildout, zope_interface, six
+, tox, zope_testing, zope_testrunner, coverage#, zc_recipe_testrunner
+}:
+
+buildPythonPackage rec {
+  pname = "martian";
+  version = "1.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "b4b48f3004f9b30789ef4fba7bb5748ff0522a80d9906b78e61fa7d1ae060883";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+    zc_buildout
+    zope_interface
+    six
+  ];
+
+  checkInputs = [
+    #zc_recipe_testrunner # Missing dependency blocked on 'zc.recipe.egg'
+    zope_testing
+    tox
+    coverage
+    zope_testrunner
+  ];
+  checkPhase = ''
+    tox
+  '';
+  doCheck = false;
```

Comment:
I'm not sure there's a huge amount of value including the details for a non-working `checkPhase`.

FWIW, it's rare that we actually want to run `tox`. In most cases when the test runner appears to be `tox` it's just wrapping the command we _actually_ want to run and getting in the way trying to do its own environment management into the deal.

e.g. in this case, looking into their `tox.ini`, I (haven't tried this but) suspect the command we actually want to run is something like `python -m zope.testrunner --test-path=src`

Reply:
Thanks for clarifying how I should be testing with `zope.testrunner`. I'm not a Python programmer. So I'm trying my best here. I can remove the `checkPhase`.

Reply:
The most blocking dependency I found was `zc.recipe.egg`. `setuptools` moved away from `eggs` in favor of `pip`. So it is no longer working. This basically blocked my attempt of getting tests running.

Reply:
`martian` tests is blocked on `zc_recipe_testrunner` which then is blocked on `zc.recipe.egg`. I have lost hope on fixing these tests.

Reply:
Can I just comment the checkPhase + checkInputs as it is? I find it valid in case of someone making an attempt of fixing it. It offers a reasonable scaffold for someone that knows fixing it. And I can have feedback from it as well, as you just did.

My preference would be to comment instead of remove.